index.js

```diff
@@ -265,18 +269,36 @@ FacebookInsightStream.prototype._collect = function ( metrics, item, buffer, eve
                 }
             }
         })
+        .catch( SkipedError, function ( error ) {
+            console.log( "facebook-insights skiped error", error );    
+        })
         .then( function () {
+            // remove the current paramater when done
+            var _metricIdx = metrics.indexOf( _metric );
+            var _evIdx = events.indexOf( _ev );
+            metrics.splice( _metricIdx, 1 );
+            events.splice( _evIdx, 1 );
+
             return this._collect( metrics, item, buffer, events );
         })
         .catch( function ( error ) {
-            if ( error.skip ) {
-                return this._collect( metrics, item, buffer )
-            } else {
-                this.emit( "error", error )
-            }
+            var retry = this._collect.bind( this, metrics, item, buffer, events );
+            return this.handleError( error, retry );
         })
 }
 
+FacebookInsightStream.prototype.handleError = function ( error, retry ) {
+    if ( error.retry === true ) {
+        return retry();
+    } else {
+        this.emit( 'error', error );
+    }
+}
+
+function SkipedError ( error ) {
+    return error.skip === true;
+}
```

Comment:
I understand that this function is used as an error constructor, what i don't really understand is where is it used? where is it instantiated? who is throwing this error that should be caught in the promise chain?